addons/survey/static/src/js/survey_timer.js

```diff
@@ -28,6 +33,12 @@ publicWidget.registry.SurveyTimerWidget = publicWidget.Widget.extend({
         var self = this;
         return this._super.apply(this, arguments).then(function () {
             self.countDownDate = moment.utc(self.timer).add(self.timeLimitMinutes, 'minutes');
+            // If user's system time is de-synchronized before the survey is started, we apply the
+            // difference in timer (if time difference is more than 5 seconds) so that we can
+            // display the 'absolute' counter
```

Comment:
I would add this in the docstring of the method as it is important.

I would also explain that this can happen if the end-user clock OR the system clock is de-synchronized.